src/posix/platform/trel.cpp

```diff
@@ -47,6 +47,7 @@
 
 #include "radio_url.hpp"
 #include "system.hpp"
+#include "common/clearable.hpp"
```

Comment:
Suggest not using this header in platform code. This is a core-internal (not a public) header and definition.

Platform layer can access any public/platform OT API/type and header files, but using core-internal headers/types creates dependency.

I know there are some situations where core-internal headers were used in `posix` platform (often done as a shortcut/hack), but I think we should aim to remove these. Suggest not adding more dependency.

Reply:
Sounds good. Using memset instead.